File: lc_go/lcGo/strings/countPrefixes2255.go
package main

import "fmt"

func countPrefixes(words []string, s string) int {
	ans := 0

	for _, word := range words {
		if len(word) > len(s) {
			continue
		}
		isPrefix := true
		for i := 0; i < len(word); i++ {
			if word[i] != s[i] {
				isPrefix = false
				break
			}
		}
		if isPrefix {
			ans += 1
		}
	}
	return ans
}

func main() {
	words := []string{"a", "b", "c", "ab", "bc", "abc"}
	s := "abc"
	res := countPrefixes(words, s)
	fmt.Println(res)

}

File: lc_cpp/lcCpp/strIngs/countPrefixes2255.cpp
#include <algorithm>
#include <string>
#include <iostream>
#include <vector>
using namespace std;
class count_prefixes2255 {
public:
    int countPrefixes(vector<string>& words,string s) {
        int ans = 0;
        for(auto const& str : words) {
            if(str.size() > s.size()) {
                continue;
            }
            bool isPrefix = true;
            for(int i = 0;i < str.size();++i) {
                // 连续子字符串
                if (s[i] - str[i] != 0) {
                    isPrefix = false;
                    break;
                }
            }
            if (isPrefix) {
                ++ans;
            }
        }
        return ans;
    }
    int countPrefixed(vector<string>& words,string s) {
        return ranges::count_if(words,[&](auto& word) {
            return s.starts_with(word);
        });
    }
};

int main() {
    count_prefixes2255 solution;
    vector<string> words = {"a","b","c","ab","bc","abc"};
    string s = "abc";
    std::cout << solution.countPrefixed(words,s) << std::endl;
    return 0;
}
